src/osrm/osrm.cpp

```diff
@@ -15,7 +16,18 @@ namespace osrm
 
 // Pimpl idiom
 
-OSRM::OSRM(engine::EngineConfig &config) : engine_(std::make_unique<engine::Engine>(config)) {}
+OSRM::OSRM(engine::EngineConfig &config)
+{
+    if (engine::Engine<engine::algorithm::CoreCH>::CheckCompability(config))
```

Comment:
At the moment this makes a priority of check CoreCH -> CH.
Is it ok to add a command line option in this PR to check requested algorithm  `config.use_core_ch && engine::Engine<engine::algorithm::CoreCH>::CheckCompability(config)`?
It will unlock possibility to use both CH and MLD in one dataset.